### DISCUSSION TITLE
Where should I store secrets in an Android app?

### DISCUSSION THREAD

**Discussion by eamonburns** [fyne-io/fyne#4739] [Q&A] [answered]
I am pretty new to this, and I have come across an issue. I need to store a database access token, and database URL for an app running on Android. How would I go about doing this?

**Comment by andydotxyz** [comment]
We don't currently have secure storage support, but the Preferences api keeps the data in your app sandbox.

**Reply by eamonburns** [reply]
Ok, so that isn't encrypted, but it shouldn't be accessible by anything else?

**Reply by eamonburns** [reply]
Looking at the documentation shows this https://docs.fyne.io/api/v2.4/preferences.html
but all that is is an interface. Would I have to implement the storage myself?

**Reply by andydotxyz** [reply, marked as answer]
No, the interface is the definition, you get the implementation by calling `App.Preferences()`.
I'd recommend reading these docs: https://docs.fyne.io/explore/preferences

**Reply by eamonburns** [reply]
Ah, this makes more sense. Thanks!
